url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/visualisation/charts/Generic_LineGraph.java

```diff
@@ -0,0 +1,561 @@
+                col - length, row - length,
+                col + length, row + length);
+        draw(line);
+        line = new Line2D.Double(
+                col - length, row + length,
+                col + length, row - length);
+        draw(line);
+    }
+
+    public static void main(String[] args) {
+        Generic_Visualisation.getHeadlessEnvironment();
+
+        /*
+         * Initialise title and File to write image to
+         */
+        String title;
+        File file;
+        String format = "PNG";
+        if (args.length != 2) {
+            System.out.println(
+                    "Expected 2 args:"
+                    + " args[0] title;"
+                    + " args[1] File."
+                    + " Recieved " + args.length + " args.");
+            // Use defaults
+            title = "Example Line Graph";
+            System.out.println("Use default title: " + title);
+            file = new File(
+                    new File(System.getProperty("user.dir")),
+                    title.replace(" ", "_") + "." + format);
+            System.out.println("Use default File: " + file.toString());
+        } else {
+            title = args[0];
+            file = new File(args[1]);
+        }
+        int dataWidth = 500;
+        int dataHeight = 250;
+        String xAxisLabel = "X";
+        String yAxisLabel = "Y";
+        boolean drawOriginLinesOnPlot = true;
+        int barGap = 1;
+        int xIncrement = 1;
+        int numberOfYAxisTicks = 11;
+        BigDecimal yMax;
+        yMax = null;
+        BigDecimal yPin = BigDecimal.ZERO;
+        BigDecimal yIncrement = BigDecimal.ONE;
+        //int yAxisStartOfEndInterval = 60;
+        int decimalPlacePrecisionForCalculations = 10;
+        int decimalPlacePrecisionForDisplay = 3;
+        RoundingMode roundingMode = RoundingMode.HALF_UP;
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
+        Generic_LineGraph chart = new Generic_LineGraph(
+                executorService,
+                file,
+                format,
+                title,
+                dataWidth,
+                dataHeight,
+                xAxisLabel,
+                yAxisLabel,
+                yMax,
+                yPin,
+                yIncrement,
+                numberOfYAxisTicks,
+                decimalPlacePrecisionForCalculations,
+                decimalPlacePrecisionForDisplay,
+                roundingMode);
+        chart.setData(chart.getDefaultData());
+        chart.run();
+        Future future = chart.future;
+        Generic_Execution.shutdownExecutorService(
+                executorService, future, chart);
+    }
+
+    @Override
+    public Dimension draw() {
+        drawOutline();
+        drawTitle(getTitle());
+        drawAxes();
+        drawData();
+        drawLegend();
+        Dimension newDim = new Dimension(getImageWidth(), getImageHeight());
+        return newDim;
+    }
+
+    public void drawAxes() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 47 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 322-341, lines 344-352 into separate methods.